pkg-manager/core/src/install/index.ts

```diff
@@ -481,80 +526,87 @@ Note that in CI environments, this setting is enabled by default.`,
         if (Object.values(ctx.projects).some((project) => pkgHasDependencies(project.manifest))) {
           throw new Error(`Headless installation requires a ${WANTED_LOCKFILE} file`)
         }
+        return null
+      }
```

Comment:
One thing I like about `tryFrozenInstall()` now being its own function is that we can early return, making logic more linear instead of a deeply nested series of if-else chains.